aks-node-controller/parser/helper_test.go

```diff
@@ -17,14 +17,130 @@ limitations under the License.
 package parser
 
 import (
+	_ "embed"
 	"encoding/base64"
+	"encoding/json"
 	"reflect"
 	"testing"
 
 	"github.com/Azure/agentbaker/aks-node-controller/helpers"
 	aksnodeconfigv1 "github.com/Azure/agentbaker/aks-node-controller/pkg/gen/aksnodeconfig/v1"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/google/go-cmp/cmp"
 )
 
+var expectedKubeletConfigFlags = "--address=0.0.0.0" +
+	" --anonymous-auth=false" +
+	" --authentication-token-webhook=true" +
+	" --authorization-mode=Webhook" +
+	" --cgroups-per-qos=true" +
+	" --client-ca-file=/etc/kubernetes/certs/ca.crt" +
+	" --cluster-dns=10.0.0.10" +
+	" --cluster-domain=cluster.local" +
+	" --enforce-node-allocatable=pods" +
+	" --event-qps=0" +
+	" --eviction-hard=memory.available<750Mi,nodefs.available<10%,nodefs.inodesFree<5%" +
+	" --feature-gates=RotateKubeletServerCertificate=true,DynamicKubeletConfig=false" +
+	" --image-gc-high-threshold=85" +
+	" --image-gc-low-threshold=80" +
+	" --kube-reserved=cpu=100m,memory=1638Mi" +
+	" --max-pods=110" +
+	" --node-status-update-frequency=10s" +
+	" --pod-manifest-path=/etc/kubernetes/manifests" +
+	" --pod-max-pids=-1" +
+	" --protect-kernel-defaults=true" +
+	" --read-only-port=10255" +
+	" --resolv-conf=/etc/resolv.conf" +
+	" --rotate-certificates=true" +
+	" --rotate-server-certificates=true" +
+	" --streaming-connection-idle-timeout=4h0m0s" +
+	" --system-reserved=cpu=2,memory=1Gi" +
+	" --tls-cert-file=/etc/kubernetes/certs/kubeletserver.crt" +
```

Comment:
hypernit: --tls-private-key-file would also have been passed by the bootstrapper if --tls-cert-file is there, thus I'd probably add it here too to make this more representative of production

Reply:
fyi, the scripts embedded on the VHD will filter these flags out during CSE before starting kubelet when kubelet serving certificate rotation is enabled

Reply:
oh I'm blind, it's already there :D